unittest/frames.cpp

```diff
@@ -96,5 +104,92 @@ BOOST_AUTO_TEST_CASE ( test_jacobian )
+  Model::Index parent_idx = model.existJointName("rarm2_joint")?model.getJointId("rarm2_joint"):(Model::Index)(model.njoints-1);
+  const std::string & frame_name = std::string( model.names[parent_idx]+ "_frame");
+  const SE3 & framePlacement = SE3::Random();
+  model.addFrame(Frame (frame_name, parent_idx, 0, framePlacement, OP_FRAME));  
+  se3::Data data(model);
+  se3::Data data_ref(model);
+  
+  VectorXd q = randomConfiguration(model, -1 * Eigen::VectorXd::Ones(model.nq), Eigen::VectorXd::Ones(model.nq) );
+  VectorXd v = VectorXd::Random(model.nv);
+  VectorXd a = VectorXd::Random(model.nv);
+  
+  computeJointJacobiansTimeVariation(model,data,q,v);
+  forwardKinematics(model,data_ref,q,v,a);
+  framesForwardKinematics(model,data_ref);
+  framesForwardKinematics(model,data);
+
+  BOOST_CHECK(isFinite(data.dJ));
+
+  Model::Index idx = model.getFrameId(frame_name);
+  const Frame & frame = model.frames[idx];
+  BOOST_CHECK(frame.placement.isApprox_impl(framePlacement));
+    
+  
+  Data::Matrix6x J(6,model.nv); J.fill(0.);
+  Data::Matrix6x dJ(6,model.nv); dJ.fill(0.);
+  
+  // Regarding to the world origin
+  getFrameJacobian<WORLD>(model,data,idx,J);
+  getFrameJacobianTimeVariation<WORLD>(model,data,idx,dJ);
+  
+  Motion v_idx(J*v);
+  const Motion & v_ref_local = frame.placement.actInv(data_ref.v[parent_idx]);
+  const Motion & v_ref = data_ref.oMf[idx].act(v_ref_local);
+  BOOST_CHECK(v_idx.isApprox(v_ref));
+  
+  Motion a_idx(J*a + dJ*v);
+  const Motion & a_ref_local = frame.placement.actInv(data_ref.a[parent_idx]);
+  const Motion & a_ref = data_ref.oMf[idx].act(a_ref_local);
+  BOOST_CHECK(a_idx.isApprox(a_ref));
+  
+  
+  // Regarding to the local frame
+  getFrameJacobian<LOCAL>(model,data,idx,J);
+  getFrameJacobianTimeVariation<LOCAL>(model,data,idx,dJ);
+  
+  v_idx = (Motion::Vector6)(J*v);
+  BOOST_CHECK(v_idx.isApprox(v_ref_local));
+              
+  a_idx = (Motion::Vector6)(J*a + dJ*v);
+  BOOST_CHECK(a_idx.isApprox(a_ref_local));
+  
+  // compare to finite differencies
+  {
+    Data data_ref(model), data_ref_plus(model);
+    
+    const double alpha = 1e-8;
+    Eigen::VectorXd q_plus(model.nq);
+    q_plus = integrate(model,q,alpha*v);
+    
+    Data::Matrix6x J_ref(6,model.nv); J_ref.fill(0.);
+    computeJointJacobians(model,data_ref,q);
+    framesForwardKinematics(model,data);
+    getFrameJacobian<WORLD>(model,data_ref,idx,J_ref);
+    
+    Data::Matrix6x J_ref_plus(6,model.nv); J_ref_plus.fill(0.);
+    computeJointJacobians(model,data_ref_plus,q_plus);
+    framesForwardKinematics(model,data);
+    getFrameJacobian<WORLD>(model,data_ref_plus,idx,J_ref_plus);
+
+    Data::Matrix6x dJ_ref(6,model.nv); dJ_ref.fill(0.);
+    dJ_ref = (J_ref_plus - J_ref)/alpha;
+    
+    computeJointJacobiansTimeVariation(model,data,q,v);
+    Data::Matrix6x dJ(6,model.nv); dJ.fill(0.);
+    getFrameJacobianTimeVariation<WORLD>(model,data,idx,dJ);
+
+    //BOOST_CHECK(dJ.isApprox(dJ_ref,sqrt(alpha)));
```

Comment:
Where is the test here?

Reply:
The comment was a mistake. I have uncommented the BOOST_CHECK.